crates/oxc_ecmascript/src/to_int_32.rs

```diff
@@ -2,79 +2,177 @@
 ///
 /// [ToInt32]: https://tc39.es/ecma262/#sec-toint32
 ///
-/// This is copied from [Boa](https://github.com/boa-dev/boa/blob/61567687cf4bfeca6bd548c3e72b6965e74b2461/core/engine/src/builtins/number/conversions.rs)
+/// This is copied from [Boa](https://github.com/boa-dev/boa/blob/95c8d4820ad10ce32892dd75673b1d8b8854f974/core/engine/src/builtins/number/conversions.rs)
 pub trait ToInt32 {
     fn to_int_32(&self) -> i32;
 }
 
 impl ToInt32 for f64 {
-    #[expect(clippy::float_cmp, clippy::cast_possible_truncation, clippy::cast_possible_wrap)]
     fn to_int_32(&self) -> i32 {
-        const SIGN_MASK: u64 = 0x8000_0000_0000_0000;
-        const EXPONENT_MASK: u64 = 0x7FF0_0000_0000_0000;
-        const SIGNIFICAND_MASK: u64 = 0x000F_FFFF_FFFF_FFFF;
-        const HIDDEN_BIT: u64 = 0x0010_0000_0000_0000;
-        const PHYSICAL_SIGNIFICAND_SIZE: i32 = 52; // Excludes the hidden bit.
-        const SIGNIFICAND_SIZE: i32 = 53;
-
-        const EXPONENT_BIAS: i32 = 0x3FF + PHYSICAL_SIGNIFICAND_SIZE;
-        const DENORMAL_EXPONENT: i32 = -EXPONENT_BIAS + 1;
-
-        fn is_denormal(number: f64) -> bool {
-            (number.to_bits() & EXPONENT_MASK) == 0
+        #[cfg(target_arch = "aarch64")]
+        {
+            if std::arch::is_aarch64_feature_detected!("jscvt") {
```

Comment:
`error: unknown aarch64 target feature: jscvt`
The full list of valid params can be found here: https://doc.rust-lang.org/beta/std/arch/macro.is_aarch64_feature_detected.html